src/test/java/in/org/projecteka/hiu/consent/ConsentUserJourneyTest.java

```diff
@@ -111,6 +124,154 @@ public void shouldThrowInsertionError() throws JsonProcessingException {
                 .is5xxServerError();
     }
 
+    @Test
+    public void shouldCreateConsentArtefacts() throws JsonProcessingException {
+        ConsentArtefactResponse consentArtefactResponse = consentArtefactResponse().build();
+        var consentArtefactResponseJson = new ObjectMapper().writeValueAsString(consentArtefactResponse);
+
+        consentManagerServer.enqueue(new MockResponse()
+                .setHeader("Content-Type", "application/json")
+                .setBody(consentArtefactResponseJson));
+
+        String consentRequestId = "consent-request-id-1";
+        ConsentNotificationRequest consentNotificationRequest = consentNotificationRequest()
+                .consentRequestId(consentRequestId)
+                .consents(singletonList(consentArtefactReference().build()))
+                .build();
+        ConsentRequest consentRequest = consentRequest()
+                .id(consentRequestId)
+                .patient(consentArtefactPatient().id("5@ncg").build())
+                .build();
+
+        when(consentRepository.get(eq(consentRequestId)))
+                .thenReturn(Mono.create(consentRequestMonoSink -> consentRequestMonoSink.success(consentRequest)));
+
+        when(consentRepository.insertConsentArtefact(eq(consentArtefactResponse.getConsentDetail())))
+                .thenReturn(Mono.create(MonoSink::success));
+
+        webTestClient
+                .post()
+                .uri("/consent/notification/")
+                .header("Authorization", "bmNn")
+                .contentType(MediaType.APPLICATION_JSON)
+                .bodyValue(consentNotificationRequest)
+                .accept(MediaType.APPLICATION_JSON)
+                .exchange()
+                .expectStatus()
+                .isOk();
+    }
+
+    @Test
+    public void shouldReturn404OnNotificationWhenConsentRequestNotFound() {
+        String consentRequestId = "consent-request-id-1";
+        ConsentNotificationRequest consentNotificationRequest = consentNotificationRequest()
+                .consentRequestId(consentRequestId)
+                .consents(singletonList(consentArtefactReference().build()))
+                .build();
+
+        when(consentRepository.get(eq(consentRequestId)))
+                .thenReturn(Mono.create(consentRequestMonoSink -> consentRequestMonoSink.success(null)));
+
+        webTestClient
+                .post()
+                .uri("/consent/notification/")
+                .header("Authorization", "bmNn")
+                .contentType(MediaType.APPLICATION_JSON)
+                .bodyValue(consentNotificationRequest)
+                .accept(MediaType.APPLICATION_JSON)
+                .exchange()
+                .expectStatus()
+                .is4xxClientError();
+    }
+
+    @Test
+    public void shouldReturn404OnNotificationWhenConsentRequestCouldNotBeFetched() {
```

Comment:
It should not return 404 if there was a failure when tried to read from DB, rather it should return 500 as we were not sure that whether it was because of the request-id, connection failure or something else. What do you think?

Reply:
Test name is wrong in this. It is actually returning 500 only.